data/Builder.go

```diff
@@ -0,0 +1,110 @@
+package data
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+//Datum represent one data point
+type Datum map[string]interface{}
+
+//Dataset represents an array of data points
+type Dataset []Datum
```

Comment:
I would call this either `Data`, `DataArray`, or `DataList`.  Data is the plural of Datum, after all.  Also, it isn't an Set (which is unordered and guarantees uniqueness).